src/components/KaitoModels.tsx

```diff
@@ -242,7 +242,29 @@ const KaitoModels = () => {
   }, []);
   function handleDeploy(model: PresetModel) {
     const yamlString = generateWorkspaceYAML(model);
-    const parsedYaml = yaml.load(yamlString);
+
+    // For multi-document YAML, we need to handle it differently
+    // Split the YAML string by document separator and parse the first document
+    const yamlDocuments = yamlString.split('---\n').filter(doc => doc.trim());
```

Comment:
I don't think you need to parse this back and forth between string and object. `EditorDialog`'s items prop accepts an array of object as argument so I'd simply pass whatever we get from `generateWorkspace` function to https://github.com/kubernetes-sigs/headlamp/blob/main/frontend/src/components/common/Resource/EditorDialog.tsx#L80

Reply:
![image](https://github.com/user-attachments/assets/9a5c1690-129e-4e2a-bd44-23e40b799796)

I think excluding the secret would be the best choice here. We at least need to provide a valid YAML for users